src/PeriodEnd/Refunds/SFA.DAS.ProviderPayments.Calc.Refunds/Services/LearnerBuilder.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Domain;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Dto;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Infrastructure.Repositories;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Services.Dependencies;
+using SFA.DAS.ProviderPayments.Calc.Shared.Infrastructure.Data;
+
+namespace SFA.DAS.ProviderPayments.Calc.Refunds.Services
+{
+    public class LearnerBuilder : ILearnerBuilder
+    {
+        private readonly IRequiredPaymentRepository _requiredPaymentRepository;
+        private readonly IHistoricalPaymentsRepository _historicalPaymentsRepository;
+
+        public LearnerBuilder(
+            IRequiredPaymentRepository requiredPaymentRepository,
+            IHistoricalPaymentsRepository historicalPaymentsRepository)
+        {
+            _requiredPaymentRepository = requiredPaymentRepository;
+            _historicalPaymentsRepository = historicalPaymentsRepository;
+        }
+
+        public List<LearnerData> CreateLearnersForProvider(long ukprn)
```

Comment:
A suggestion - if LearnerData has a domain model for one data type (`HistoricalPayment`) but a data entity for the other (`RequiredPaymentEntity`) then it is inconsistent and should use all domain models.

Reply:
Agreed, but again lets talk to Carl. This again is to do with inconsistent styles. And allowing the use of entities in the domain